numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -2844,15 +2844,23 @@ OBJECT_nonzero (PyObject **ip, PyArrayObject *ap)
         if (*ip == NULL) {
             return NPY_FALSE;
         }
-        return (npy_bool) PyObject_IsTrue(*ip);
+        int istrue = PyObject_IsTrue(*ip);
+        if (istrue == -1) {
+            return NPY_FALSE;
+        }
+        return (npy_bool) istrue;
     }
     else {
         PyObject *obj;
         memcpy(&obj, ip, sizeof(obj));
         if (obj == NULL) {
             return NPY_FALSE;
         }
-        return (npy_bool) PyObject_IsTrue(obj);
+        int istrue = PyObject_IsTrue(obj);
+        if (istrue == -1) {
+            return NPY_FALSE;
```

Comment:
Let's not do this.  If anything, the old version is better.

The caller currently needs to always check if an error occurred. Returning `(npy_bool)-1` might give us a way to improve that.